### PR TITLE
Add enrollments to sessions

### PR DESCRIPTION
**Pull request by alxdarksage**
BRIDGE-3097

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/models/accounts/StudyParticipantTest.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public void canSerializeForCache() throws Exception {
 
         assertEquals(node.get("attributes").get("A").textValue(), "B");
         assertEquals(node.get("attributes").get("C").textValue(), "D");
-        assertEquals(node.size(), 29);
+        assertEquals(node.size(), 30);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/models/accounts/StudyParticipantTest.java:125]: Test that it's in the JSON as you'd expect.

File: src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionInfoTest.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ public void userSessionInfoSerializesCorrectly() throws Exception {
         assertEquals(consentStatus.size(), 7);
         
         // ... and no things that shouldn't be there
-        assertEquals(node.size(), 24);
+        assertEquals(node.size(), 25);
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionInfoTest.java:109]: Add tests that it's in the JSON you'd expect.

File: src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java [hunk 3]
```diff
@@ -53,7 +68,8 @@ public void canSerialize() throws Exception {
             .withHealthCode("healthCode")
             .withSharingScope(ALL_QUALIFIED_RESEARCHERS)
             .withRoles(ImmutableSet.of(ADMIN))
-            .withDataGroups(ImmutableSet.of("group1", "group2")).build();
+            .withDataGroups(ImmutableSet.of("group1", "group2"))
+            .withEnrollments(enrollments).build();
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java:70]: Again, more tests of the JSON

File: src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        
+        JsonNode node = BridgeObjectMapper.get().valueToTree(detail);
+        
+        assertEquals(node.get("externalId").textValue(), TEST_EXTERNAL_ID);
+        assertTrue(node.get("consentRequired").booleanValue());
+        assertEquals(node.get("enrolledOn").textValue(), CREATED_ON.toString());
+        assertNull(node.get("enrolledBySelf"));
+        assertNull(node.get("withdrawnBySelf"));
+        assertEquals(node.get("type").textValue(), "EnrollmentInfo");
+    }
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java:47]: Test deserialization

File: src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        JsonNode node = BridgeObjectMapper.get().valueToTree(detail);
+        
+        assertEquals(node.get("externalId").textValue(), TEST_EXTERNAL_ID);
+        assertTrue(node.get("consentRequired").booleanValue());
+        assertEquals(node.get("enrolledOn").textValue(), CREATED_ON.toString());
+        assertNull(node.get("enrolledBySelf"));
+        assertEquals(node.get("withdrawnOn").textValue(), MODIFIED_ON.toString());
+        assertNull(node.get("withdrawnBySelf"));
+        assertEquals(node.get("type").textValue(), "EnrollmentInfo");
+    }
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java:84]: Test deserialization

Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java:84]: Also look at UserSessionInfoTest to verify it is copied correctly.

File: src/test/java/org/sagebionetworks/bridge/models/accounts/StudyParticipantTest.java (hunk 1 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/accounts/StudyParticipantTest.java:125]: Added

File: src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionInfoTest.java (hunk 2 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionInfoTest.java:109]: Added

File: src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java:70]: In this case, we test that the study participant is equals in the newly deserialized session, and this includes the enrollments.

File: src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java:47]: Added

File: src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java (hunk 5 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/studies/EnrollmentInfoTest.java:84]: Done

File: src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java (hunk 3 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/models/accounts/UserSessionTest.java:70]: This isn't exposed through the API, UserSessionInfo is the object that we return from the API. In this test, we already verify the participant is serialized and deserialized correctly for storage in Redis.

**Review by jnelsonjava — APPROVED** [review]
Looks good.

**Pull request merged**